src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -499,6 +499,53 @@ internal void Insert(ref T[] list, string? name, MemberListType listType)
                     }
                 }
 
+                private void MergeWithGlobalListInOrder(T[] list)
+                {
+                    T[]? cachedMembers = m_allMembers;
+
+                    if (cachedMembers == null)
+                    {
+                        m_allMembers = list;
+                        return;
+                    }
+
+                    int length = list.Length;
+                    T[] newCache = list;
+
+                    // When all members are requested there should be no element in cache that not found within all members list.
+                    // But somehow in very rare case such element found in cache and not appending that to the all members list
+                    // causing random test failure, must be related to the bug commented on row 597
+                    Array.Resize(ref newCache, length + 1);
```

Comment:
Why the +1?

Reply:
Technically there should be no any extra element, but were random test failure in random projects, I could not really get clear repro or that I could figure out what is going on, so basically followed the existing logic
https://github.com/buyaa-n/runtime/blob/227d5645fbafc1b79189c0757b2525a13bac60b4/src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs#L591-L602